### PR TITLE
scrm-39 -  Replace X templates on SubPanels with Smarty

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

I order to make the sub panels fit the SuiteP design. So that the column names are at the top. I have change the sub panels so that they now support smarty templates. 

This means you can also override the **include/SubPanel/tpls/SubPanelDynamic.tpl** file in the themes. Which could be quite useful. I have create a new view called **include/ListView/ListViewSubPanel.php** which is extends the standard ListView.

 I haven't completely gotten rid of the x templates. I have left some of the custom views which currently are not being used anyway. I have also left some hard coded stuff like the pagination in the ListViewSubpanel. As some of the processing is handed here. Perhaps in the future we might move some of this code out into a template.

## Motivation and Context

- Makes the subpanels fit the SuiteP design.
- Keeps the Suite7 and SuiteR the same.

## How To Test This

- Check each sub panel in SuiteCRM.
- Ensure that each function works like it did before.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: include/ListView/ListViewSubPanel.php [hunk 1]
```diff
@@ -0,0 +1,890 @@
+<?php
+    if (!defined('sugarEntry') || !sugarEntry) {
```
Comment (samus-aran) [inline include/ListView/ListViewSubPanel.php]: Move to below the Licence

File: include/ListView/ListViewSubPanel.php [hunk 2]
```diff
@@ -0,0 +1,890 @@
...
+        die('Not A Valid Entry Point');
+    }
+    /*********************************************************************************
+     * SugarCRM Community Edition is a customer relationship management program developed by
```
Comment (samus-aran) [inline include/ListView/ListViewSubPanel.php]: Update License

**Comment by samus-aran** [comment]
Tested on Chrome/Firefox

- When reducing the window size the subpanel fields are not be subsequently hidden similar to the ListView.  When window is reduced the fields in the subpanel should be hidden depending on how many can fix in the screen size.

**Comment by daniel-samson** [comment]
Closing for staging. Please refer to https://github.com/salesagility/SuiteCRM/pull/2657

**Pull request closed without merging**